### PR TITLE
Command line interface

### PR DESCRIPTION
**Pull request by cancerberoSgx**
Hey, your lib is awesome, I created a Command line interface app, documented it, write options types, basic tests, removed png.js files and used npm install png.js. 

The only thing missing is that you publish the library to npm (npm publish) so the cli app depends on that (currently is requiring the project relatively and that won't work for users npm-installing it. 

Please tell me if you think this PR could be merged (after review) since I need it published on npm so users can easily install it. Thanks!

### CODE REVIEW

**Comment by jankovicsandras** [comment]
Hi Sebastián,

Thanks for your CLI app, it looks great! I didn't have time to test it yet, but skimmed some parts, and it will be much more useful and safer than the current nodecli example.

However, I need some time to think about merging this PR. Some features I would like to keep in the ImageTracer projects are: dependency free, single file, batteries included, KISS.
Yes, it seems I contradict myself here as imagetracerjs includes a copy of PNG.js files, but it's meant to be more like an example. I don't want to be opiniated here, I encourage users to find the best image loader / Node.js canvas implementation. (Random real, not sarcasm question here: can PNG.js load JPG, GIF, etc. ? I don't know.) But I want to be able to download imagetracerjs.zip from GitHub and have it work out-of-the box in 10 years.
Adding these dependencies (glob, minimist, png.js) might be in conflict with these ideas, it's hard to decide what's best.

I recommend you to fork the project and publish it with your CLI app on NPM. You can do whatever you want as it's Unlicense / Public domain. :)

Best wishes,
András

**Comment by cancerberoSgx** [comment]
> However, I need some time to think about merging this PR. Some features I would like to keep in the ImageTracer projects are: dependency free, single file, batteries included, KISS.

I agree, I was thinking about the same thing after integrating Jimp to support several input formats: https://github.com/cancerberoSgx/imagetracerjs/blob/a02d8a6d3188a84830d80d607d2f6906d3c67920/imagetracer_v1.2.5.js#L63 - I just killed the project simplicity :)

But as you can notice, the folder cli has its own package.json, is not a dependency nor it doesn't depend on files outside the folder so it can be considered an independent project. Ideally , people use mono-repos for this, or even separate repositories but I didn't want to take the credit for such a tool being you the author. 

Also notice that cli/ is mainly a Command line Interface - if I just remove its index.js (so it doesn't export any js API) then it's only usable as a command line tool. 

Besides the CLI I think it would be also good to have: 

 * test (in node and in browser)
 * a basic playground where users can change options / load images online to understand the library

And for both things/projects using image reader implementation will be needed. 

In summary, do you think I should create a separate project for this? I do think having "consumer" projects like these will add great value and help the library to mature. But if you think otherwise feel free to close this PR and if I have your permission I can move the cli part to a separate project.

Also a comment, I see there are several methods referencing the DOM, and I think embedding / loading images in the DOM should not be part of the core, but a separate project (DOM users should know how to do this). 

Thanks, keep it up!

**Comment by cancerberoSgx** [comment]
Although I'm not sure is the project needs canvas and the DOM, just in case, if you are using it just for loading the image's bitmap, that can be done without involving the DOM and using the same API both for node and browser by using libraries like 'node-fetch' or 'buffer'. I just discovered these and for this kind of library that aims to support both browser and node it can be of great help. 

```js
const blobToBuffer = require('blob-to-buffer')
const fetch = require('node-fetch'); // it uses window.fetch in browser and pollyfill in node 
var option = {
    method: 'GET',
    headers: headers,
    mode: 'cors',
    cache: 'default'
  }
  const response = await fetch(url, options)
  const blob = await response.blob()
  const buffer = await blobToBuffer(blob)
  new PNGReader(content).parse((error, png) => {
      if (error) {       
      } else {
       const outputContent = ImageTracer.imagedataToSVG({ ...png, data: png.pixels }, options)
      }
    })
```
